index.bs

```diff
@@ -204,27 +204,14 @@ partial interface Element {
 Its value is read by the <a>report image element timing</a> and the <a>report text element timing</a> algorithms.
 
-Every image resource that is fetched from a URL has an associated <a>image request</a>, which is a <a>fetch request</a>.
-The request has an <dfn>associated element</dfn>, the {{Element}} causing this image resource.
-
-NOTE: we assume that there is one <a>image request</a> for each {{Element}} that a <a>background-image</a> property affects and for each URL that the <a>background-image</a> property specifies. So, for example, if there is a style with two URLs affecting all <{div}>s, and there are two <{div}>s, then there will be four <a>image requests</a>. This means that a single <a>background-image</a> property could produce multiple {{PerformanceElementTiming}} entries because it can affect multiple elements and because it can specify multiple URLs.
+Every {{Element}} has an <dfn>associated image request</dfn> which is initially null.
+When either of {{HTMLImageElement}}, {{SVGImageElement}}, or {{HTMLVideoElement}} require a new image resource (to be displayed as an image or poster image), its <a>associated image request</a> is set to a new <a>image request</a> caused by the new resource.
 
 Note: Every image resource that is obtained from a URL whose <a spec=url>scheme</a> is equal to "data" has an associated <a>image request</a> which is not fetched but still needs to be loaded and decoded.
-This request also has an <a>associated element</a> which caused this image resource.
-
-Each {{Document}} has <dfn>images pending rendering</dfn>, a list of <a>image requests</a> which are considered loaded and decoded but not yet rendered.
-When an <a>image request</a> has become <a>completely available</a> and decoded, run the following steps:
+This request can be the <a>associated image request</a> of an {{Element}}.
 
-<div algorithm="image element loaded">
-    1. Let |imageRequest| be the input <a>image request</a>.
-    1. Let |element| be |imageRequest|'s <a>associated element</a>.
-    1. If |imageRequest| is not a data URL [[RFC2397]] and if the <a>timing allow check</a> fails for |imageRequest|'s resource, then:
-        1. Run the <a>report image element timing</a> algorithm, passing in |imageRequest| and 0.
-    1. Otherwise, run the following steps:
-        1. Let |root| be |element|'s <a for="tree">root</a>.
-        1. If |root| is not a {{Document}}, return.
-        1. Add |imageRequest| to |root|'s <a>images pending rendering</a>.
-</div>
+Each {{Document}} has <dfn>images pending rendering</dfn>, a list of triples ({{Element}}, <a>image request</a>, DOMHighResTimeStamp) which are considered loaded and decoded but not yet rendered.
+When an {{Element}}'s <a>associated image request</a> has become <a>completely available</a> and decoded, run the algorithm to <a>process an image that finished loading</a> passing in the {{Element}} and its <a>associated image request</a> as inputs.
```

Comment:
This should be part of HTML's image processing model (e.g. a step that adds that should be invoked from https://html.spec.whatwg.org/multipage/images.html#updating-the-image-data:img-all-3 as well as other places that set the "completely available" state, or from the adding of the image to the "list of available images", or from the invoked onload event).

Also, you should drop "decoded" here, as I don't think we want to expose decoding time for cross-origin no-TAO case.

Also also, what happens with images the never become "completely available"? Do we not report them? If so, worth a note

Reply:
Again, do we want to make this specific to \<img\> and ignore the others? I think an image can only be rendered when it is completely available, so yea we do not report images that are "Broken", which is the only other possible final state.

Reply:
Yeah, good point about this not being HTMLImageElement specific. I think we can:
* Land the current language as is.
* PR the HTML spec, wrapping addition of elements to the list of available images with a function/algorithm
* Add a step in that algorithm that triggers this.
* Open an issue on the HTML spec saying that HTMLVideoElement, SVG images, form field images and probably a few more should all go through the list of available images.